src/web/js/output-ui.js

```diff
@@ -401,25 +401,30 @@
             console.error("maybeLocToAST should not be passed a builtin source location.", loc);
             return runtime.ffi.makeNone();
           },
-          "srcloc": function(filename, start_line, start_col, _, end_line, end_col, _) {
+          "srcloc": function(filename, start_line, start_col, _, end_line, end_col, __) {
             var prelude = ""
             for(var i=1; i < start_line; i++) {prelude += "\n";}
             for(var i=0; i < start_col; i++)  {prelude += " "; }
             if(!documents.has(filename))
               return runtime.ffi.makeNone();
-            var source = documents.
-              get(filename).
-              getRange(
-                new CodeMirror.Pos(start_line - 1, start_col),
-                new CodeMirror.Pos(  end_line - 1,   end_col));
+            var start = new CodeMirror.Pos(start_line - 1, start_col);
+            var   end = new CodeMirror.Pos(  end_line - 1,   end_col);
+            var source = documents.get(filename).getRange(start, end);
             runtime.pauseStack(function(restarter) {
-              runtime.run(function(_, __) {
+              runtime.runThunk(function() {
                 return runtime.getField(PP, "surface-parse").app(prelude + source, filename);
-              }, runtime.namespace, {
-                sync: false
               }, function(result) {
                 if(runtime.isSuccessResult(result)) {
-                  restarter.resume(runtime.ffi.makeSome(result.result.dict.block.dict.stmts.dict.first));
+                  try {
+                    restarter.resume(runtime.ffi.makeSome(result.result.dict.block.dict.stmts.dict.first));
```

Comment:
Technically, not quite right: this will also catch failures in the subsequent resumption and claim they were the fault of `makeSome(result....first)`.  Better to use a local variable here just around the field extraction, or better yet, just use a chained 
```
var res = result;
res = res && res.result;
res = res && getField(res, "block");
res = res && getField(res, "stmts");
res = res && getField(res, "first");
if (res) { restarter.resume(makeSome(res)); }
else { console.error ... }
```